apache/couchdb-fauxton
# Upgrade to webpack 4

## Overview

Upgrade to webpack 4 and latest version of the webpack loaders.

Updates to webpack configurations:
 - MiniCssExtractPlugin replaced the ExtractTextPlugin
 - Options for UglifyJsPlugin, CommonsChunkPlugin, DefinePlugin and
   LimitChunkCountPlugin were removed since webpack 4 now provides
   the same features as built-in options.

## Testing recommendations

- Verify dev server works
  - Run `npm run dev` and verify dev server works
  - Change a JS file and verify webpack recompiles the bundles

- Verify production bundles are created correctly
  - Run `npm run webpack:release`
  - Verify all assets are generated under the `dist` folder. E.g.:

```
$ ls -R dist/release
dashboard.assets index.html

dist/release/dashboard.assets:
css   fonts img   js

dist/release/dashboard.assets/css:
styles.3653d66aac9e267d78bb.css

dist/release/dashboard.assets/fonts:
fauxtonicon5.eot         fauxtonicon5.woff        fontawesome-webfont.ttf
fauxtonicon5.ttf         fontawesome-webfont.eot  fontawesome-webfont.woff

dist/release/dashboard.assets/img:
CouchDB-negative-logo.png couchdb-logo.png          github.png                loader.gif
asf-feather-logo.png      fauxtonicon5.svg          googleplus.png            twitter.png
couch-watermark.png       fontawesome-webfont.svg   linkedin.png

dist/release/dashboard.assets/js:
bundle.3653d66aac9e267d78bb.js   manifest.2dd37e048a9a76292451.js vendor.d5ab0d6762c52f2419e6.js
```
